### PR TITLE
Registration class

### PR DESCRIPTION
**Pull request by jennyfolkesson**
I turned the grid registration functions into a class for improved ease of use and readability.

### CODE REVIEW

**Comment by jennyfolkesson** [comment]
FYI: codcov check failed because coverage has gone down compared to master, but that's only because I removed the tests directory from being counted as covered code.

File: array_analyzer/transform/array_generation.py [hunk 1]
```diff
@@ -84,10 +82,10 @@ def get_spot_intensity(coords, im, background, params, search_range=2):
...
     """
     # values in mm
-    spot_width = params['spot_width']
-    pix_size = params['pixel_size']
-    n_rows = params['rows']
-    n_cols = params['columns']
+    spot_width = constants.params['spot_width']
+    pix_size = constants.params['pixel_size']
+    n_rows = constants.params['rows']
+    n_cols = constants.params['columns']
```
Comment (smguo) [inline array_analyzer/transform/array_generation.py:88]: 👍

File: array_analyzer/transform/point_registration.py [hunk 2]
```diff
@@ -95,157 +64,230 @@ def icp(source, target, max_iterate=50, matrix_diff=1.):
...
+        grid_cols = grid_cols.flatten()
+        grid_rows = grid_rows.flatten()
+        grid_coords = np.vstack([grid_rows.T, grid_cols.T]).T
+        return grid_coords
+
+    def create_gaussian_particles(self,
+                                  mean_point=(0, 0),
+                                  scale_mean=1.,
+                                  angle_mean=0.):
+        """
```
Comment (smguo) [inline array_analyzer/transform/point_registration.py:134]: This is a minor point but is there a reason to make the mean of the gaussian a class method input, but the std a constant in `constant`? And it seems to me the default values for mean will always be used? Maybe consider moving the mean to `constant` as well?

File: tests/transform/point_registration_test.py [hunk 3]
```diff
@@ -1,15 +1,37 @@
...
+import array_analyzer.extract.constants as constants
 import array_analyzer.transform.point_registration as registration
 
-im_shape = (50, 75)
-reg_coords = np.ones((5, 2), dtype=np.float32)
 
+@pytest.fixture
+def register_inst():
+    im_shape = (50, 100)
+    spot_coords = np.zeros((10, 2), dtype=np.float32)
```
Comment (smguo) [inline tests/transform/point_registration_test.py]: If I understand this correctly, you set up the test to register 3 fiducials to 10 spots that are all (0, 0)? Can the fiducials actually register to these spots? Maybe consider setting up the spots in a way that they can be registered by the fiducials, so if some future changes break the registration the test would catch it?

**Review by smguo — APPROVED** [review]
I tested the PR and it ran without issues. I only have some minor comments.

File: array_analyzer/transform/point_registration.py (hunk 2 above)
Comment (jennyfolkesson) [reply array_analyzer/transform/point_registration.py:134]: Good point, I'll move it to constants.

File: tests/transform/point_registration_test.py (hunk 3 above)
Comment (jennyfolkesson) [reply tests/transform/point_registration_test.py]: Good suggestion, thanks!

**Pull request merged**